src/messages/test_proposals.rs

```diff
@@ -128,6 +130,123 @@ fn proposal_queue_functions() {
+                .unwrap();
+        let bob_credential_bundle =
+            CredentialBundle::new("Bob".into(), CredentialType::Basic, ciphersuite.name()).unwrap();
+
+        // Generate KeyPackages
+        let alice_key_package_bundle =
+            KeyPackageBundle::new(&[ciphersuite.name()], &alice_credential_bundle, Vec::new())
+                .unwrap();
+        let bob_key_package_bundle =
+            KeyPackageBundle::new(&[ciphersuite.name()], &bob_credential_bundle, Vec::new())
+                .unwrap();
+        let bob_key_package = bob_key_package_bundle.key_package();
+        let alice_update_key_package_bundle =
+            KeyPackageBundle::new(&[ciphersuite.name()], &alice_credential_bundle, Vec::new())
+                .unwrap();
+        let alice_update_key_package = alice_update_key_package_bundle.key_package();
+        assert!(alice_update_key_package.verify().is_ok());
+
+        let group_context = GroupContext {
+            group_id: GroupId::random(),
+            epoch: GroupEpoch(0),
+            tree_hash: vec![],
+            confirmed_transcript_hash: vec![],
+        };
+
+        // Let's create some proposals
+        let add_proposal_alice1 = AddProposal {
+            key_package: alice_key_package_bundle.key_package().clone(),
+        };
+        let add_proposal_alice2 = AddProposal {
+            key_package: alice_key_package_bundle.key_package().clone(),
+        };
+        let add_proposal_bob1 = AddProposal {
+            key_package: bob_key_package.clone(),
+        };
+
+        let proposal_add_alice1 = Proposal::Add(add_proposal_alice1);
+        let proposal_reference_add_alice1 =
+            ProposalReference::from_proposal(ciphersuite, &proposal_add_alice1);
+        let proposal_add_alice2 = Proposal::Add(add_proposal_alice2);
+        let proposal_add_bob1 = Proposal::Add(add_proposal_bob1);
+
+        // Frame proposals in MLSPlaintext
+        let mls_plaintext_add_alice1 = MLSPlaintext::new(
+            LeafIndex::from(0u32),
+            &[],
+            MLSPlaintextContentType::Proposal(proposal_add_alice1.clone()),
+            &alice_credential_bundle,
+            &group_context,
+        );
+        let mls_plaintext_add_alice2 = MLSPlaintext::new(
+            LeafIndex::from(1u32),
+            &[],
+            MLSPlaintextContentType::Proposal(proposal_add_alice2),
+            &alice_credential_bundle,
+            &group_context,
+        );
+        let mls_plaintext_add_bob1 = MLSPlaintext::new(
+            LeafIndex::from(1u32),
+            &[],
+            MLSPlaintextContentType::Proposal(proposal_add_bob1.clone()),
+            &alice_credential_bundle,
+            &group_context,
+        );
+
+        // This should set the order of the proposals.
+        let proposals = &[
+            &mls_plaintext_add_alice1,
+            &mls_plaintext_add_alice2,
+            &mls_plaintext_add_bob1,
+        ];
```

Comment:
Can you specify what the expected outcome is, when you add the same proposal twice?

Reply:
I don't think adding the same proposal twice does anything here test-wise. I've removed it, as that part of the functionality is already covered in the other test.